src/lib.rs

```diff
@@ -1531,10 +1529,29 @@ fn encode_block(seq: &Sequence, fi: &FrameInvariants, fs: &mut FrameState,
         TxType::DCT_DCT
     };
 
-    // if inter, predict here
-    //mode.predict(&mut rec.mut_slice(po), tx_size);
-
     if is_inter {
+        // Inter mode prediction can take place once for a whole partition,
+        // instead of each tx-block.
+        for p in 0..3 {
```

Comment:
In the AV1 spec, the compute_prediction() function has a loop from 0 to  1 + HasChroma * 2. Presumably there are cases where a small partition may not have associated chroma?

Reply:
Yes, you are right. If luma is 4x4, then there is no chroma 4x4 until the last 4x4 luma.
Also, in that case, the offset for chroma planes should be derived from the 1st 4x4 luma's coordinate. Thanks for pointing this out! I will fix soon.